Mathlib/Algebra/Order/Monoid/Lemmas.lean

```diff
@@ -249,6 +251,14 @@ theorem mul_right_cancel'' [ContravariantClass α α (swap (· * ·)) (· ≤ ·
 
 end PartialOrder
 
+section LinearOrder
+variable [LinearOrder α] {a b c d : α} [CovariantClass α α (· * ·) (· < ·)]
+  [CovariantClass α α (swap (· * ·)) (· < ·)]
+
+@[to_additive] lemma min_le_max_of_mul_le_mul (h : a * b ≤ c * d) : min a b ≤ max c d :=
+by simp_rw [min_le_iff, le_max_iff]; contrapose! h; exact mul_lt_mul_of_lt_of_lt h.1.1 h.2.2
```

Comment:
I think the style in mathlib4 is that the `by` goes on the line before, like this:
```lean
@[to_additive] lemma min_le_max_of_mul_le_mul (h : a * b ≤ c * d) : min a b ≤ max c d := by 
  simp_rw [min_le_iff, le_max_iff]; contrapose! h; exact mul_lt_mul_of_lt_of_lt h.1.1 h.2.2
```
But other than that this looks fine.

Reply:
Whoops, sorry.